backend/routes/api/users.js

```diff
@@ -0,0 +1,166 @@
+router.get("/:user_id", async (req, res) => {
+  const userId = req.params.user_id;
+  const user = await User.findById(userId);
+  res.json({
+    code: 200,
+    result: user,
+    success: true
+  });
+});
+
+// create new user
+router.post(
+  "/",
+  celebrate({
+    body: Joi.object().keys({
+      firstName: Joi.string().required(),
+      lastName: Joi.string().required(),
+      oauthId: Joi.string().required(),
+      propicUrl: Joi.string().required(),
+      isApproved: Joi.boolean().default(false),
+      role: Joi.string().required(),
+      location: Joi.string().required()
+    })
+  }),
+  async (req, res) => {
+    const data = req.body;
+    const newUser = new User({
+      firstName: data.firstName,
+      lastName: data.lastName,
+      oauthId: data.oauthId,
+      propicUrl: data.propicUrl,
+      role: data.role,
+      location: data.location
+    });
+    await newUser.save();
+    res.json({
+      code: 200,
+      message: "User Successfully Created",
+      success: true
+    });
+  }
+);
+
+// set role
+router.put(
+  "/setRole/:user_id",
+  celebrate({
+    body: Joi.object().keys({
+      firstName: Joi.string(),
+      lastName: Joi.string(),
+      oauthId: Joi.string(),
+      propicUrl: Joi.string(),
+      isApproved: Joi.boolean(),
+      role: Joi.string().required(),
+      location: Joi.string()
+    })
+  }),
+  async (req, res) => {
+    const data = req.body;
+    const userId = req.params.user_id;
+
+    const user = await User.findByIdAndUpdate(
+      userId,
+      { $set: { role: data.role } },
+      { new: true }
+    );
+    const ret = user
+      ? {
+          code: 200,
+          message: "User Role Updated Successfully",
+          success: true
+        }
+      : {
+          code: 404,
+          message: "User Not Found",
+          success: false
+        };
+    res.json(ret);
+  }
+);
+
+// approve user
+router.put(
+  "/approve/:user_id",
+  celebrate({
+    body: Joi.object().keys({
+      firstName: Joi.string(),
+      lastName: Joi.string(),
+      oauthId: Joi.string(),
+      propicUrl: Joi.string(),
+      isApproved: Joi.boolean(),
+      role: Joi.string(),
+      location: Joi.string()
+    })
+  }),
+  async (req, res) => {
+    const userId = req.params.user_id;
+
+    const user = await User.findByIdAndUpdate(
+      userId,
+      { $set: { isApproved: true } },
+      { new: true }
+    );
+    const ret = user
+      ? {
+          code: 200,
+          message: "User Approved Successfully",
+          success: true
+        }
+      : {
+          code: 404,
+          message: "User Not Found",
+          success: false
+        };
+    res.json(ret);
+  }
+);
+
+// delete user
+router.delete("/:user_id", async (req, res) => {
+  const userId = req.params.user_id;
+  const user = await User.findByIdAndRemove(userId);
+  const ret = user
+    ? {
+        code: 200,
+        message: "User deleted successfully",
+        success: true
+      }
+    : {
```

Comment:
Same here, let's just make sure we set `res.status(404)` on the error. Feel free to switch to using `Hapi.js` [Boom](https://github.com/hapijs/boom), made by the same authors of celebrate.js. It standardizes the errors and handles these for you. This isn't a high priority so it's not a blocking change, but we'll probably want to switch using `Boom` at some point in the future to make sure everything is standard.